app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>  
+<% if policy(@user).show? %>
```

Comment:
Call the new method than `.show` to authorize viewing another user's profile that you are not authorized to view.

Reply:
+1